crm_claim_code/__init__.py

```diff
@@ -15,11 +16,9 @@ def create_code_equal_to_id(cr):
 
 
 def assign_old_sequences(cr, registry):
-    claim_obj = registry['crm.claim']
-    sequence_obj = registry['ir.sequence']
-    claim_ids = claim_obj.search(cr, SUPERUSER_ID, [], order="id")
-    for claim_id in claim_ids:
-        cr.execute('UPDATE crm_claim '
-                   'SET code = \'%s\' '
-                   'WHERE id = %d;' %
-                   (sequence_obj.get(cr, SUPERUSER_ID, 'crm.claim'), claim_id))
+    with Environment.manage():
+        env = Environment(cr, SUPERUSER_ID, {})
+
+        claims = env['crm.claim'].search([], order="id")
+        for claim in claims:
+            claim.code = env['ir.sequence'].next_by_code('crm.claim')
```

Comment:
Why don't you store now the sequence handler in `sequence_obj` variable like before, so you avoid to search for it on each loop pass?